tests/phpunit/Config/ApiTest.php

```diff
@@ -144,42 +120,96 @@ public function testCreateAuthQuery(): void
         ];
 
         $api = new Api(new NullLogger(), $apiConfig, ['key' => 'val'], []);
-        $request = new Request('GET', 'http://example.com?foo=bar');
-        $restClientMock = $this->getClientMock($request);
-        /** @var RestClient $restClientMock */
-        $api->getAuth()->authenticateClient($restClientMock);
-        self::assertEquals(['foo' => 'bar', 'param' => 'val'], $request->getQuery()->toArray());
+        $history = new HistoryContainer();
+        $restClient = RestClientMockBuilder::create()
+            ->addResponse200('{"foo": "bar"}')
+            ->setHistoryContainer($history)
+            ->setInitCallback(function (RestClient $client) use ($api): void {
+                $api->getAuth()->attachToClient($client);
+            })
+            ->getRestClient();
+
+        self::assertEquals(
+            (object) ['foo' => 'bar'],
+            $restClient->download($restClient->createRequest(['endpoint' => 'http://example.com?foo=bar']))
+        );
+
+        $request = $history->shift()->getRequest();
         self::assertInstanceOf(Query::class, $api->getAuth());
+        self::assertEquals(['foo' => 'bar', 'param' => 'val'], Psr7Query::parse($request->getUri()->getQuery()));
     }
 
     public function testCreateAuthOAuth20Bearer(): void
```

Comment:
Tu su niektore testy duplicitne, a testuju to co sa testuje napr. v `OAuth20Test`,  .... len trocha inak, ...ale nechal som ich.